src/query_parser/parser.y

```diff
@@ -786,6 +823,17 @@ query ::= expr(A) ARROW LSQB vector_query(B) RSQB . { // main parse, hybrid quer
   QueryNode_AddChild(B, A);
 }
 
+query ::= text_expr(A) ARROW LSQB vector_query(B) RSQB . { // main parse, hybrid query as entire query case.
```

Comment:
Not covered by previous rules?
```
expr(A) ::= text_expr(B). [TEXTEXPR] {
query ::= expr(A) ARROW LSQB vector_query(B) RSQB . {
```

Reply:
apparently not. got syntax error on `hello=>[KNN...]`

Reply:
Again because of shift on `text_expr`